textscreen/txt_fileselect.c

```diff
@@ -512,6 +512,43 @@ int TXT_CanSelectFiles(void)
     return ZenityAvailable();
 }
 
+/*
+ * ExpandExtension
+ * given an extension (like wad)
+ * return a pointer to a string that is a case-insensitive
+ * pattern representation (like [Ww][Aa][Dd])
+ */
+static char * ExpandExtension(char *orig)
+{
+    int oldlen,newlen, i;
+    char *c,*newext = NULL;
+
+    oldlen = strlen(orig);
+    newlen = oldlen * 4; // pathalogical case: 'w' => '[Ww]'
+    newext = (char *)malloc(newlen+1);
+    if (newext)
```

Comment:
Better to check for the error case, eg.

    if (newext == NULL)
    {
        return NULL;
    }